File: tournaments/nontransitiveDice/nontransitiveDice.py
def nontransitiveDice(dice):
    q=[0,0,0]
    def k(r,s):
        t = 0
        for i in range(6):
            for j in range(6):
                if dice[r][i]>dice[s][j]:t+=1
                if dice[r][i]<dice[s][j]:t-=1
        if t>0:q[r]+=1
        if 0>t:q[s]+=1
    k(0,1)
    k(1,2)
    k(2,0)
    return q.count(1) > 1

File: tournaments/nontransitiveDice/nontransitiveDice.js
function nontransitiveDice(dice) {
    const C = function(a, b) {
        let n = 0;
        a.map(x => b.map(y => n += (x > y) - (x < y)));
        return (n > 0) - (0 > n);
    };
    const [a,b,c] = [...dice];
    
    return !!(C(a,b) - C(a,c) & C(b,a) - C(b,c) & C(c,a) - C(c,b));
}
